src/execute.c

```diff
@@ -1088,13 +1092,21 @@ int jq_compile_args(jq_state *jq, const char* str, jv args) {
 
     nerrors = builtins_bind(jq, &program);
     if (nerrors == 0) {
-      nerrors = block_compile(program, &jq->bc, locations);
+      program = block_drop_unreferenced(program);
+      /*
+       * We could dump_block(program) here, but block_compile() modifies
+       * the program it in important ways, so we leave it to
+       * block_compile() to dump it.
+       */
+      int dump = jv_equal(jv_true(), jv_object_get(jv_copy(opts), jv_string("dump_block")));
```

Comment:
It would probably be useful to gather into an object the pre- and post-`block_compile()` block representation of the jq program.  `block_compile()` does make some transformations on the input `block`, and it does do things like set `bytecode_pos`.  It is useful to be able to observe these transformations.